keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,298 @@
+            shape=(None,), dtype=tf.int32, name="input_mask"
+        )
+
+        # Embed tokens and positions.
+        embedding_layer = TokenAndPositionEmbedding(
+            vocabulary_size=vocabulary_size,
+            sequence_length=max_sequence_length,
+            embedding_dim=hidden_dim,
+            embeddings_initializer=_roberta_kernel_initializer(),
+            name="embeddings",
+        )
+        embedding = embedding_layer(token_id_input)
+
+        # Sum, normalize and apply dropout to embeddings.
+        x = keras.layers.LayerNormalization(
+            name="embeddings_layer_norm",
+            axis=-1,
+            epsilon=1e-5,  # Original paper uses this epsilon value
+            dtype=tf.float32,
+        )(embedding)
+        x = keras.layers.Dropout(
+            dropout,
+            name="embeddings_dropout",
+        )(x)
+
+        # Apply successive transformer encoder blocks.
+        for i in range(num_layers):
+            x = TransformerEncoder(
+                num_heads=num_heads,
+                intermediate_dim=intermediate_dim,
+                activation="gelu",
+                dropout=dropout,
+                kernel_initializer=_roberta_kernel_initializer(),
+                name=f"transformer_layer_{i}",
+            )(x, padding_mask=input_mask)
+
+        # Construct the Roberta output.
+        sequence_output = x
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "input_ids": token_id_input,
+                "input_mask": input_mask,
+            },
+            outputs={
+                "sequence_output": sequence_output,
+            },
+            name=name,
+            trainable=trainable,
+        )
+        # All references to `self` below this line
+        self.token_embedding = embedding_layer.token_embedding
+        self.vocabulary_size = vocabulary_size
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        # BOS token '<s>' is equivalent to '[CLS]' from BERT
+        self.start_token_index = start_token_index
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "dropout": self.dropout,
+                "start_token_index": self.start_token_index,
+            }
+        )
+        return config
+
+
+# TODO: add RobertaMLM, different from BERT.
```

Comment:
We can remove this TODO as we don't necessarily plan to support pretraining for every model.